pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -853,6 +873,46 @@ func formatDomainCPUTune(vmi *v1.VirtualMachineInstance, domain *Domain, c *Conv
 	return nil
 }
 
+func appendDomainIOThreadPin(domain *Domain, thread uint, cpuset string) {
+	iothreadPin := CPUTuneIOThreadPin{}
+	iothreadPin.IOThread = thread
+	iothreadPin.CPUSet = cpuset
+	domain.Spec.CPUTune.IOThreadPin = append(domain.Spec.CPUTune.IOThreadPin, iothreadPin)
+}
+
+func formatDomainIOThreadPin(vmi *v1.VirtualMachineInstance, domain *Domain, c *ConverterContext) error {
+	iothreads := int(domain.Spec.IOThreads.IOThreads)
+	vcpus := int(calculateRequestedVCPUs(vmi))
+
+	if iothreads >= vcpus {
```

Comment:
OK, so there's steam coming out my ears. :) Let me recap what I think this is doing:
If threads is greater than or equal to VCPU's, just round robin threads to CPUs--basically the best we can do.
If we have more VCPU's than threads, let's not waste them! instead we'll allocate multiple CPUs to a given IOThread. Since there's no rule that says that VCPU's needs to be a multiple of threads, `series` will basically count how many threads get an extra VCPU. `remainder` is basically a calculation of the number of CPUs each thread is going to get (but the first `series` threads get an extra one).

If you agree with my impression, then I agree that's what this block of code does.

Reply:
I've updated the logic a bit. Hope it's clearer now.